server/controllers/finance/journal/index.js

```diff
@@ -193,6 +194,42 @@ function find(options) {
 }
 
 /**
+* journalEntryList
+* Allows you to select which transactions to print
+*/
+function journalEntryList(options) { 
+  let uuids =  options.uuids.map(function(uuid) {
+    return db.bid(uuid);
+  });
+
+  let sql = `
+    SELECT BUID(p.uuid) AS uuid, p.project_id, p.fiscal_year_id, p.period_id,
+      p.trans_id, p.trans_date, BUID(p.record_uuid) AS record_uuid,
+      dm1.text AS hrRecord, p.description, p.account_id, p.debit, p.credit,
+      p.debit_equiv, p.credit_equiv, p.currency_id, c.name AS currencyName,
+      BUID(p.entity_uuid) AS entity_uuid, em.text AS hrEntity,
+      BUID(p.reference_uuid) AS reference_uuid, dm2.text AS hrReference,
+      p.comment, p.origin_id, p.user_id, p.cc_id, p.pc_id, pro.abbr,
+      pro.name AS project_name, per.start_date AS period_start,
+      per.end_date AS period_end, a.number AS account_number, u.display_name
+    FROM posting_journal p
+      JOIN project pro ON pro.id = p.project_id
+      JOIN period per ON per.id = p.period_id
+      JOIN account a ON a.id = p.account_id
+      JOIN user u ON u.id = p.user_id
+      JOIN currency c ON c.id = p.currency_id
+      LEFT JOIN entity_map em ON em.uuid = p.entity_uuid
+      LEFT JOIN document_map dm1 ON dm1.uuid = p.record_uuid
+      LEFT JOIN document_map dm2 ON dm2.uuid = p.reference_uuid
+    WHERE p.uuid IN (?)
+    ORDER BY p.trans_date DESC
```

Comment:
I think you should first order by `trans_date` and then by `record_uuid`.  This will make sure the transactions stay in groups.